lib/internal/util.js

```diff
@@ -366,6 +366,43 @@ function spliceOne(list, index) {
   list.pop();
 }
 
+let getStructuredStack;
+function isInsideNodeModules() {
+  if (!getStructuredStack) {
+    // Lazy-load to avoid a circular dependency.
+    const { runInNewContext } = require('vm');
+    // Use `runInNewContext()` to get something tamper-proof and
+    // side-effect-free. Since this is currently only used for a deprecated API,
+    // the perf implications should be okay.
+    getStructuredStack = runInNewContext('(' + function() {
+      Error.prepareStackTrace = function(err, trace) {
+        err.stack = trace;
+      };
+      Error.stackTraceLimit = Infinity;
+
+      return function structuredStack() {
+        // eslint-disable-next-line no-restricted-syntax
+        return new Error().stack;
+      };
+    } + ')()', {}, { filename: 'structured-stack' });
+  }
+
+  const stack = getStructuredStack();
+
+  // Iterate over all stack frames and look for the first one not coming
+  // from inside Node.js itself:
+  for (const frame of stack) {
+    const filename = frame.getFileName();
+    // If a filename does not start with / or contain \,
+    // it's likely from Node.js core.
+    if (!/^\/|\\/.test(filename))
+      continue;
+    return /[\\/]node_modules[\\/]/.test(filename);
```

Comment:
if require.main.filename contains `node_modules`, we could ignore that `node_modules` folder here. I know plenty of devs (including myself), that use a `node_modules` folder to contain all the dev code to make cross module development easier.

Something like:

``` js
const nodeModulesParent = require.main.filename.indexOf('/node_modules/') // add windows support
function isNodeModules (filename) {
  const i = filename.lastIndexOf('/node_modules/')
  return i !== -1 && i > nodeModulesParent
}
```

This has the added benefit of making deprecations show if you cd into node_modules and run a test fx (i do that a bunch as well)

Reply:
@mafintosh thanks, i'll look into it later, this makes sense!

I think we might adjust it abit because there might be different prefixes before `/node_modules/` and we want to take care of backlashes as well, but generally this sgtm

Reply:
@mafintosh updated, please take a look!